nevergrad/optimization/optimizerlib.py

```diff
@@ -762,6 +763,94 @@ def _internal_provide_recommendation(self) -> ArrayLike:
         return self.avg
 
 
+@registry.register
+class Splitter(base.Optimizer):
+    """Combines optimizers, each of them working on their own variables.
+    
+    num_optims: number of optimizers
+    num_vars: number of variable per optimizer.
+
+    E.g. for 5 optimizers, each of them working on 2 variables, we can use:
+    opt = Splitter(instrumentation=10, num_workers=3, num_optims=5, num_vars=[2, 2, 2, 2, 2])
+    or equivalently:
+    opt = Splitter(instrumentation=10, num_workers=3, num_vars=[2, 2, 2, 2, 2])
+    Given that all optimizers have the same number of variables, we can also do:
+    opt = Splitter(instrumentation=10, num_workers=3, num_optims=5)
+
+    This is 5 parallel (by num_workers = 5).
+
+    Be careful! The variables refer to the deep representation used by optimizers.
+    For example, a categorical variable with 5 possible values becomes 5 continuous variables.
+    """
+
+    def __init__(self, instrumentation: Union[int, Instrumentation], budget: Optional[int] = None, num_workers: int = 1, num_optims: Optional[int] = None, num_vars: Optional[List[Any]] = None) -> None:
+        super().__init__(instrumentation, budget=budget, num_workers=num_workers)
+        if num_vars:
+            if num_optims:
+                assert num_optims == len(num_vars)
+            else:
+                num_optims = len(num_vars)
+            assert sum(num_vars) == self.dimension
+        else:
+            if not num_optims:  # if no num_vars and no num_optims, just assume 3.
+                num_optims = 2
+            # num_vars not given: we will distribute variables equally.
+        if num_optims > self.dimension:
+            num_optims = self.dimension
+            if num_vars:
+                num_vars = num_vars[:num_optims]
```

Comment:
With the asserts above, this modifies num_vars only if the number of optimizers is greater than the dimension and some elements of num_vars are 0 or negative. I think it would be better to remove this and assert that every number in num_vars is strictly positive (or positive or 0, in which case you want to keep all the non zero num_vars and not the first dimension ones).

Reply:
you're right, i now assert that the user provides a meaningful num_vars with everything fine.